src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx

```diff
@@ -56,67 +53,48 @@ const title = i18n.translate('indexPatternManagement.indexPatternTable.title', {
 
 interface Props extends RouteComponentProps {
   canSave: boolean;
+  showCreateDialog?: boolean;
 }
 
-export const IndexPatternTable = ({ canSave, history }: Props) => {
+export const IndexPatternTable = ({
+  history,
+  canSave,
+  showCreateDialog: showCreateDialogProp = false,
+}: Props) => {
   const {
     setBreadcrumbs,
     uiSettings,
     indexPatternManagementStart,
     chrome,
+    data,
     docLinks,
-    application,
     http,
-    data,
+    notifications,
+    application,
   } = useKibana<IndexPatternManagmentContext>().services;
   const [indexPatterns, setIndexPatterns] = useState<IndexPatternTableItem[]>([]);
-  const [creationOptions, setCreationOptions] = useState<IndexPatternCreationOption[]>([]);
-  const [sources, setSources] = useState<MatchedItem[]>([]);
-  const [remoteClustersExist, setRemoteClustersExist] = useState<boolean>(false);
-  const [isLoadingSources, setIsLoadingSources] = useState<boolean>(true);
   const [isLoadingIndexPatterns, setIsLoadingIndexPatterns] = useState<boolean>(true);
+  const [showCreateDialog, setShowCreateDialog] = useState<boolean>(showCreateDialogProp);
 
   setBreadcrumbs(getListBreadcrumbs());
   useEffect(() => {
     (async function () {
-      const options = await indexPatternManagementStart.creation.getIndexPatternCreationOptions(
-        history.push
-      );
       const gettedIndexPatterns: IndexPatternTableItem[] = await getIndexPatterns(
         uiSettings.get('defaultIndex'),
-        indexPatternManagementStart,
         data.indexPatterns
       );
-      setIsLoadingIndexPatterns(false);
-      setCreationOptions(options);
       setIndexPatterns(gettedIndexPatterns);
+      setIsLoadingIndexPatterns(false);
+      if (gettedIndexPatterns.length === 0) {
+        setShowCreateDialog(true);
+      }
     })();
-  }, [history.push, indexPatterns.length, indexPatternManagementStart, uiSettings, data]);
-
-  const removeAliases = (item: MatchedItem) =>
-    !((item as unknown) as ResolveIndexResponseItemAlias).indices;
-
-  const loadSources = () => {
-    getIndices(http, () => [], '*', false).then((dataSources) =>
-      setSources(dataSources.filter(removeAliases))
-    );
-    getIndices(http, () => [], '*:*', false).then((dataSources) =>
-      setRemoteClustersExist(!!dataSources.filter(removeAliases).length)
-    );
-  };
-
-  useEffect(() => {
-    getIndices(http, () => [], '*', false).then((dataSources) => {
-      setSources(dataSources.filter(removeAliases));
-      setIsLoadingSources(false);
-    });
-    getIndices(http, () => [], '*:*', false).then((dataSources) =>
-      setRemoteClustersExist(!!dataSources.filter(removeAliases).length)
-    );
-  }, [http, creationOptions]);
+  }, [indexPatternManagementStart, uiSettings, data]);
 
   chrome.docTitle.change(title);
 
+  const isRollup = new URLSearchParams(useLocation().search).get('type') === 'rollup';
```

Comment:
this looks like a hack, I'd look for an alternative. maybe there is a way to pass current state as a prop

Reply:
The router doesn't parse query params so this code is going to exist somewhere, its just a question of where. I think it makes sense to locate it near the code that opens the flyout but I'm open to other suggestions. 

It should also be noted that in the long run `rollup` specific index pattern creation code is likely to be removed. At least once the next version of rollups is released.